### PR TITLE
Feature/skale 1709 bounty

### PR DESCRIPTION
**Pull request by DimaStebaev**
Add bounty distribution feature.

### CODE REVIEW

**Review by kladkogex — APPROVED** [review]

File: contracts/delegation/DelegationRequestManager.sol [hunk 1]
```diff
@@ -48,6 +48,7 @@ contract DelegationRequestManager is Permissions {
         ValidatorService validatorService = ValidatorService(
             contractManager.getContract("ValidatorService")
         );
+        require(validatorService.checkValidatorExists(validatorId), "Validator does not exist");
```
Comment (yavrsky) [inline contracts/delegation/DelegationRequestManager.sol]: line 69 is the same

**Review by yavrsky — APPROVED** [review]

File: contracts/delegation/DelegationRequestManager.sol [hunk 2]
```diff
@@ -65,7 +65,13 @@ contract DelegationRequestManager is Permissions {
             delegationPeriodManager.isDelegationPeriodAllowed(delegationPeriod),
             "This delegation period is not allowed"
         );
-        require(validatorService.checkValidatorExists(validatorId), "Validator is not registered");
+        require(validatorService.checkValidatorExists(validatorId), "Validator does not exist");
+
+        // check that there is enough money
+        uint holderBalance = SkaleToken(contractManager.getContract("SkaleToken")).balanceOf(holder);
+        uint lockedToDelegate = tokenState.getLockedCount(holder) - tokenState.getPurchasedAmount(holder);
```
Comment (payvint) [inline contracts/delegation/DelegationRequestManager.sol:72]: Does it possible to overflow ?

**Review by payvint — APPROVED** [review]
It looks like true distribution!!

File: contracts/delegation/DelegationRequestManager.sol (hunk 2 above)
Comment (DimaStebaev) [reply contracts/delegation/DelegationRequestManager.sol:72]: No.

**Pull request merged**